### PR TITLE
added playlist logic and component

### PR DESCRIPTION
**Pull request by ashishnnnnn**
-> User can add playlist.
-> User can remove playlist.
-> User can add videos to playlist.
-> User can remove videos from playlist.
-> Playlist page to see all your playlist at one place.

### CODE REVIEW

File: src/Components/Videocard/Videocard.jsx [hunk 1]
```diff
@@ -103,58 +107,117 @@ export const Videocard = ({ ele }) => {
...
-                </div>
-              </li>
-            </ul>
+            <>
+              {location.substring(0, 5) === "/play" ? (
+                <ul className="flex-column gap-0-5">
+                  <li>
+                    <div className="flex mini-video-option gap-0-5">
+                      <i
+                        onClick={async () => {
```
Comment (shubhamkumar648) [inline src/Components/Videocard/Videocard.jsx]: suggestion: you can also do explicit function here

**Review by shubhamkumar648 — APPROVED** [review]
Good use of context  and  reducer, good Naming convention  ,you can merge

File: src/Reducer/UserDataReducer.js [hunk 2]
```diff
@@ -48,6 +48,50 @@ export function UserDataReducer(state, action) {
...
+      const new_playlist_after_add = action.payload;
+      localStorage.setItem(
+        "my_playlist",
+        JSON.stringify(new_playlist_after_add)
+      );
+      return { ...state, playlist: new_playlist_after_add };
+    case "ADD_VIDEO_TO_PLAYLIST":
+      const playlist_to_change = action.payload;
+
+      const new_playlist_after_SingleVideoAdd = state.playlist.map((ele) =>
```
Comment (lannisterrr) [inline src/Reducer/UserDataReducer.js]: variable name can be short and use of camel case would be ideal.

**Review by lannisterrr — COMMENTED** [review]
Clean and structured code , playlist logic is easy to understand , good work . Some suggestions made.

File: src/Components/Videocard/Videocard.jsx (hunk 1 above)
Comment (ashishnnnnn) [reply src/Components/Videocard/Videocard.jsx]: Done ..

File: src/Reducer/UserDataReducer.js (hunk 2 above)
Comment (ashishnnnnn) [reply src/Reducer/UserDataReducer.js]: Done

**Review by tushartiwari7 — APPROVED** [review]
Everything looks fine! code is neat and clean.
A small suggestion to create handler functions whenever there's more than one thing you are doing on `onClick`. Will look more structured.

**Pull request merged**